app/design/frontend/rwd/default/template/configurableswatches/catalog/product/view/type/configurable/swatch-js.phtml

```diff
@@ -13,8 +13,11 @@
  * @license     https://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
  */
 ?>
-<script type="text/javascript">
-    document.observe('dom:loaded', function() {
-        var swatchesConfig = new Product.ConfigurableSwatches(spConfig);
-    });
+<?php if (Mage::helper('configurableswatches')->getSwatchesProductJs()): ?>
```

Comment:
Instead of looping the attributes in the helper, we can check the `head` block:
```suggestion
<?php if ($this->getLayout()->getBlock('head')->getItems('skin_js/js/configurableswatches/swatches-product.js')): ?>
```

Reply:
I don't know, let you choose. @fballiano do you see any potential downsides? if for some reason theme move the library to another path?

Reply:
mmmm I'm wondering if somebody could break this new check changing the path to that file via layout xml... but it's probably a better risk to take compared to cycling the attributes like the getSwatchesProductJs() method does. it's fine by me.

Reply:
true